src/client/views/start.ts

```diff
@@ -20,7 +20,8 @@ export class Start extends AbstractView {
 		this.viewContainer.querySelector('.opening-btn-tn').addEventListener('click', this.goToWalletSelection.bind(this));
 	}
 
-	goToWalletSelection(){
+	goToWalletSelection() {
+		this.client.checkInternetConnectivity();
```

Comment:
I don't think the following functions should trigger if there is no internet found at that time. This should help smooth the UX. Where from testing, this allows the user to continue, but the requests to collect tokens have failed leaving a black screen.